episodes/04-git-changes.md

```diff
@@ -0,0 +1,310 @@
+but we haven't told Git we will want to save those changes,
+nor have we saved them.
+So let's do that now. 
+
+![](fig/rs_git_commit_message_unstaged.png){alt='RStudio screenshot showing dialogue box when attempting to commit an unstaged commit.'}
+
+Let's enter the following Commit message and click on Commit:
+
+```output
+List of project data files added
+```
+![](fig/rs_git_commit_unstaged.png){alt='RStudio screenshot showing dialogue box when attempting to commit an unstaged commit.'}
+
+Whoops:
+Git won't commit because we didn't use `git add` first.
+Let's fix that:
+
+![](fig/rs_git_add_file.png){alt='RStudio screenshot showing dialogue box when attempting to commit an unstaged commit.'}
+
+Checking the `Staged` box in the Git tab for index.qmd has "added" the file. Notice also, that the `M` has moved from the right-hand status column to the left-hand column.
+
+We can now,`Commit` the change successfully.
+
+![](fig/rs_git_commit_2.png){alt='RStudio screenshot showing dialogue box confirming commit was successful.'}
+
+Git insists that we add files to the set we want to commit
+before actually committing anything. This allows us to commit our changes in stages and capture changes in logical portions rather than
+only large batches.
+
+For example,
+suppose we're adding a few citations to relevant research to our thesis.
+We might want to commit those additions,
+and the corresponding bibliography entries,
+but *not* commit some of our work drafting the conclusion
+(which we haven't finished yet).
+
+To allow for this,
+Git has a special *staging area*
+where it keeps track of things that have been added to
+the current [changeset](../learners/reference.md#changeset)
+but not yet committed.
+
+:::::::::::::::::::::::::::::::::::::::::  callout
+
+## Staging Area
+
+If you think of Git as taking snapshots of changes over the life of a project,
+checking `Staged` (`git add`) specifies *what* will go in a snapshot
+(putting things in the staging area),
+and `Commit`(`git commit`) then *actually takes* the snapshot, and
+makes a permanent record of it (as a commit).
+If you don't have anything staged when you try to `Commit`,
+Git will prompt you to use `git commit -a` or `git commit --all`,
+which is kind of like gathering *everyone* to take a group photo!
+However, it's almost always better to
+explicitly add things to the staging area, because you might
+commit changes you forgot you made. (Going back to the group photo simile,
+you might get an extra with incomplete makeup walking on
+the stage for the picture because you used `-a`!)
+Try to stage things manually,
+or you might find yourself searching for "git undo commit" more
+than you would like!
+
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+![](fig/git-staging-area.svg){alt='A diagram showing how "git add" registers changes in the staging area, while "git commit" moves changes from the staging area to the repository'}
```

Comment:
```suggestion
![](fig/git-staging-area.svg){alt='A diagram showing how `git add` registers changes in the staging area, while `git commit` moves changes from the staging area to the repository'}
```
I'm not sure whether this will work inside an alt message, but I think it would be helpful for the image description to mark it as commands.